### PR TITLE
Space-hub

### PR DESCRIPTION
**Pull request by osbentech**
This is React group Project named Space-hub. It fetches Mission and Rocket data from the Api and displays it in the UI.
Below are the functionalities we implemented:

 - Setup the basic structure for the Missions
 - Get the data from the API & displaying it correctly
 - Saving the data in the local storage so that we don't call the API on every render
 - Setting actions to get data from the API, get data from local storage, and update the state of the action
 - Changing the styling of the button when it is clicked
 - Displaying the joined missions on the My Profile section using the filter method
 - Setup the basic structure for the rockets
 - Get the data from the API & displaying it correctly
 - Saving the data in the local storage so that we don't call the API on every render
- Setting actions to get data from the API, get data from local storage, and update the state of the reservation
- Changing the styling of the button & displaying "Reserved" tag when clicking the button
- Displaying the reserved rockets on the My profile section using the filter method

### CODE REVIEW

File: src/components/Rockets.js [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+          <span style={tagStyle} className={styles.tag}>Reserved</span>
+          {description}
+        </p>
+        <button
+          onClick={() => dispatch(updateStatus(id))}
+          type="button"
+          style={btnStyle}
+        >
+          {btnText}
+        </button>
```
Comment (Wes-Isaac) [inline src/components/Rockets.js:74]: - [ ] same as Mission.js kindly use conditional rendering.

File: src/components/Mission.js [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+      </div>
+      <div className={styles.btnContainer}>
+        <button
+          style={btnStyle}
+          type="button"
+          className={styles.missionBtn}
+          onClick={() => dispatch(updateStatus(id))}
+        >
+          {btnText}
+        </button>
```
Comment (Wes-Isaac) [inline src/components/Mission.js:78]: - [ ] The buttons are rendered correctly but in the project it's specified that we use react conditional rendering, it saves us lines of code kindly add that. 


image

File: package.json [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+    "@testing-library/user-event": "^13.5.0",
+    "axios": "^1.1.2",
+    "i": "^0.3.7",
+    "npm": "^8.19.2",
+    "prop-types": "^15.8.1",
+    "react": "^18.2.0",
+    "react-dom": "^18.2.0",
+    "react-redux": "^8.0.4",
+    "react-router-dom": "^6.4.2",
+    "react-scripts": "5.0.1",
```
Comment (Wes-Isaac) [inline package.json:18]: - [ ] Great job installing/using react-router and react-redux, however it's one of the requirement to also install redux logger kindly integrate that to your project

**Review by Wes-Isaac — CHANGES_REQUESTED** [review]
Hi @osbentech & @Peter1907 , 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights
- No linter error ✅.
- Great teamwork 👍.
- Great PR description 💯.

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
-N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
```

**Review by t-yanick — APPROVED** [review]
Hi Team, 

### Great job !!!
### To Highlight! :clap:  :green_circle: 

:heavy_check_mark:  Good commit messages
:heavy_check_mark: Clean/Organized code
:heavy_check_mark: Descriptive Readme file

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit:  
Congratulations! 🎉 

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

### N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**